doc/continuous_integration/circleci.md

```diff
@@ -0,0 +1,104 @@
+# CircleCI Documentation
```

Comment:
This file doesn't cover the topics:
- Artifacts, how is that setup and where are they saved to
- For all sections after workflows, it usually skips over explaining the breakdown of the cron jobs for nightly / dockerhub and only discusses the PR builder (e.x. jobs, executors, commands, steps)
- Matrix, for the nightly

Reply:
On this file, I think it would be useful to take a step back and not miss the forest for the trees. It does a good job at introducing some circle CI concepts (that I'd like to just be completed from my comments), but in some ways it doesn't really explain the specific navigation2 use of CircleCI; which mostly just just about the 3 workflows (what they do, when they run, why they run, their general description of steps) that is generally missing or overlooked in favor of explaining the components they're made of abstractly.